src/protocols/bdx/BdxTransferSession.cpp

```diff
@@ -848,13 +845,13 @@ void TransferSession::PrepareStatusReport(StatusCode code)
 {
     mStatusReportData.statusCode = code;
 
-    Encoding::LittleEndian::PacketBufferWriter bbuf(chip::MessagePacketBuffer::New(kStatusReportMinSize), kStatusReportMinSize);
-    VerifyOrReturn(!bbuf.IsNull());
+    Protocols::SecureChannel::StatusReport report(static_cast<uint16_t>(Protocols::SecureChannel::GeneralStatusCode::kFailure),
+                                                  static_cast<uint32_t>(Protocols::kProtocol_BDX), static_cast<uint16_t>(code));
 
-    bbuf.Put16(static_cast<uint16_t>(Protocols::SecureChannel::GeneralStatusCode::kFailure));
-    bbuf.Put32(Protocols::kProtocol_BDX);
-    bbuf.Put16(static_cast<uint16_t>(mStatusReportData.statusCode));
+    Encoding::LittleEndian::PacketBufferWriter bbuf(chip::MessagePacketBuffer::New(report.Size()), report.Size());
```

Comment:
Might be worth putting `Size()` into a variable here; it's not trivially foldable afaict, so we'll end up running it twice...

Reply:
done